lib/rules/no-multi-assign.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileoverview Rule to check use of multiple assignments per statement
+ * @author Stewart Rand
+ */
+
+"use strict";
+
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow use of multiple assignments per statement",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+        schema: []
+    },
+
+    create(context) {
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        return {
+            AssignmentExpression(node) {
+                if (["AssignmentExpression", "VariableDeclarator"].indexOf(node.parent.type) !== -1) {
+                    context.report({
+                        node,
+                        message: "Expected no more than one assignment"
```

Comment:
Personally, I think it might be clearer if the report message just states what it found, rather than what it expected (since it's possible the user meant to use something else). For example: "Unexpected chained assignment."